### ISSUE TITLE
AttributeError: 'TasmotaIrhvac' object has no attribute '__attr_preset_mode'. Did you mean: '__attr_preset_modes'?

### ISSUE THREAD

**Issue by andrewwong2000** [hristo-atanasov/Tasmota-IRHVAC#144]
Core. 2024.6.3
Supervisor. 2024.06.0
Operating System. 12.3
Frontend. 20240610.1

My Tasmota IR devices are all getting the error. Stack trace and YAML is below.

Thanks for any help to figure out why a working system now won't startup.

Climate.yaml

- platform: tasmota_irhvac
  name: Bedroom AC
  unique_id: bedroom_ac 
  command_topic: "cmnd/tasmota_ir_bedroom_01/irhvac"
  state_topic: "tele/tasmota_ir_bedroom_01/RESULT"
  temperature_sensor: sensor.4_in_1_multisensor_air_temperature_2
  protocol: "MITSUBISHI112"
  min_temp: 16 #optional - default 16 int value
  max_temp: 30 #optional - default 32 int value
  target_temp: 24 #optional - default 26 int value
  initial_operation_mode: "off" # optional - default "off" string value
  away_temp: 24 #optional - default 24 int value
  precision: 1.0 #optional - default 1.0 float value
  supported_modes:
    - heat
    - cool
    - "off" #Turns the AC off - Should be quoted and should be always included here
  supported_fan_speeds:                                                            
    - low                                                                          
    - medium           
    - max
    - auto  
  supported_swing_list:
    - "off"
    - vertical #up to down
  hvac_model: "1" # string value
  celsius_mode: "On" #optional - default "On" string value - you can remove it if you use celsius
  keep_mode_when_off: True
  ignore_off_temp: False #optional - default False boolean value

----

Stack Trace

Logger: homeassistant.components.climate
Source: helpers/entity_platform.py:600
integration: Climate (documentation, issues)
First occurred: 8:22:50 PM (3 occurrences)
Last logged: 8:22:50 PM

Error adding entity climate.bedroom_ac for domain climate with platform tasmota_irhvac
Error adding entity climate.study_ac for domain climate with platform tasmota_irhvac
Error adding entity climate.kitchen_ac for domain climate with platform tasmota_irhvac
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 600, in _async_add_entities
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 914, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1355, in add_to_platform_finish
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1000, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1126, in _async_write_ha_state
    state, attr, capabilities, shadowed_attr = self.__async_calculate_state()
                                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1063, in __async_calculate_state
    if state_attributes := self.state_attributes:
                           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 324, in __getattribute__
    return super().__getattribute__(__name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 557, in state_attributes
    data[ATTR_PRESET_MODE] = self.preset_mode
                             ^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 324, in __getattribute__
    return super().__getattribute__(__name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/functools.py", line 995, in __get__
    val = self.func(instance)
          ^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 639, in preset_mode
    return self._attr_preset_mode
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 324, in __getattribute__
    return super().__getattribute__(__name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 324, in __getattribute__
    return super().__getattribute__(__name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TasmotaIrhvac' object has no attribute '__attr_preset_mode'. Did you mean: '__attr_preset_modes'?

**Comment by confuzius0815** [comment]
Hi,
Not a solution, but a workaround:
I had the same problem and it did no longer occur with HA core 2024.6.2 after downgrading (from backup).

**Comment by andrewwong2000** [comment]
Thanks for the tip.. I tried downgrading to 2024.6.2 with no luck, and also tried to downgrade to 2024.5.5.

Will keep trying !

**Comment by confuzius0815** [comment]
Hi,
One more thing I did:
Before downgrading HA to 2024.6.2, I switched back the the latest commits (which I first thought would solve my problem). So actually now I am on 2024.6.2 with HA and on 2024.6.1 with this integration (at least I think so, do not remeber exactly my steps):
[https://github.com/hristo-atanasov/Tasmota-IRHVAC/tree/v2024.6.1](url)

Thinking about it, that commit might be the problem with HA (at least 2024.6.3)? :
[https://github.com/hristo-atanasov/Tasmota-IRHVAC/commit/de5a190852e4e9fc69d19bdcc23076b8c0946320](url)

Have a nice weekend.
